site/app/libraries/FileUtils.php

```diff
@@ -227,6 +227,18 @@ public static function getZipSize($filename) {
         return $size;
     }
     
+    public static function checkFileInZipName($filename) {
+        $zip = zip_open($filename);
+        if(is_resource(($zip))) {
+            while ($inner_file = zip_read($zip)) {
+                $fname = zip_entry_name($inner_file);
+                if(strpos($fname, '\'') !== false || strpos($fname, '\\') !== false || strpos($fname, '\"') !== false || strpos($fname, '<') !== false) {
+                    return false;
+                }
```

Comment:
Make this if statement into a function that returns true/false on good/failed file names